3ds Max/Max2Babylon/Exporter/BabylonExporter.Skeleton.cs

```diff
@@ -61,67 +61,41 @@ private List<IIGameNode> GetRevelantNodes(IIGameSkin skin)
                 return revelantNodesBySkin[skin];
             }
 
-            List<IIGameNode> revelantNodes = new List<IIGameNode>();
             List<IIGameNode> bones = GetBones(skin);
 
-            if(bones.Contains(null))
+            if (bones.Count == 0)
             {
-                RaiseError("Skin has bones that are outside of the exported hierarchy.", logRank);
-                RaiseError("The skin cannot be exported", logRank);
+                RaiseWarning("Skin has no bones.", logRank);
                 return new List<IIGameNode>();
             }
-            
-            // for each bone of the skin, add their parents in the revelantNodes list.
-            foreach (IIGameNode bone in bones)
+
+            if (bones.Contains(null))
             {
-                if (!revelantNodes.Contains(bone))
-                {
-                    revelantNodes.Add(bone);
-                    IIGameNode currentNode = bone.NodeParent;
-                    while (currentNode != null)
-                    {
-                        if (!revelantNodes.Contains(currentNode))
-                        {
-                            revelantNodes.Add(currentNode);
-                            currentNode = currentNode.NodeParent;
-                        }
-                        else // the node and its parents are already in the list
-                        {
-                            currentNode = null;
-                        }
-                    }
-                }
+                RaiseError("Skin has bones that are outside of the exported hierarchy.", logRank);
+                RaiseError("The skin cannot be exported", logRank);
+                return new List<IIGameNode>();
             }
 
-            // return an empty list if there is more than one root
-            List<IIGameNode> rootNodes = revelantNodes.FindAll(node => node.NodeParent == null);    // should contains only one node
-            if (rootNodes.Count > 1)
+            IIGameNode sharedRootNode = GetLowestCommonAncestor(bones);
+            if (sharedRootNode == null)
             {
                 string rootNames = "";
-                foreach(IIGameNode root in rootNodes)
-                {
-                    rootNames += $" {root.Name}";
-                }
                 RaiseError($"More than one root node for the skin: {rootNames}", logRank);
```

Comment:
Line 83's debug statement still doesn't seem correct

Reply:
Correct. There is also another problem, because the node hierarchy exported is currently not correct when there are gaps in the bone hierarchy.

I'm fixing this right now, but we will not test it before monday. @elpie89

Reply:
I made some tests, and everything works on my side, I tested with many different hierarchies

Reply:
Great, can you please upload the test scenes as part of this conversation? I'll approve the PR